src/Illuminate/Auth/SessionGuard.php

```diff
@@ -210,6 +210,69 @@ protected function userFromRecaller($recaller)
         return $user;
     }
 
+    /**
+     * Impersonate the given user.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable  $user
+     * @return void
+     *
+     * @throws \Illuminate\Auth\AuthenticationException
+     */
+    public function impersonate(AuthenticatableContract $user)
+    {
+        if (! $authenticated = $this->user()) {
+            throw new AuthenticationException('Cannot impersonate without a currently authenticated user.');
+        }
+
+        $this->session->put($this->getImpersonationName(), $authenticated->getAuthIdentifier());
+
+        $this->login($user);
+    }
+
+    /**
+     * Stop impersonating a user and resume the original authentication state.
+     *
+     * @return void
+     */
+    public function unpersonate()
```

Comment:
Any interest in `stopImpersonating()`?

Reply:
> Any interest in `stopImpersonating()` or `endImpersonation`?

I would love to choose stopImpersonating() instead of unpersonate() because I don’t see the vocabulary for unpersonate(). Hmm, I come with another option: stopImpersonate(). It saves more characters than stopImpersonating() although stopImpersonate() is same with stopImpersonating().

Reply:
I'll wait for additional feedback on this as well. Chose this for its character length being the same, but maybe `startImpersonating()` `stopImpersonating()` would work better like you guys mention 🙏

Reply:
what about `impersonateAs()` and `impersonateOff` ?

Reply:
I actually quite like `unimpersonate()`—even if it might not be a real word: It is short, catchy and gets its point across good enough 👍🏻